superset/db_engine_specs.py

```diff
@@ -980,19 +981,82 @@ def get_columns(
             result.append(cls._create_column_info(column, column.Column, column_type))
         return result
 
+    @classmethod
+    def _is_column_name_quoted(cls, column_name: str) -> bool:
+        """
+        Check if column name is in quotes
+        :param column_name: column name
+        :return: boolean
+        """
+        return column_name.startswith('"') and column_name.endswith('"')
+
+    @classmethod
+    def _get_fields(cls, cols: List[dict]) -> List[ColumnClause]:
+        """
+        Format column clauses where names are in quotes and labels are specified
+        :param cols: columns
+        :return: column clauses
+        """
+        column_clauses = []
+        dot_regex = r'\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)'
```

Comment:
Nit, Python supports verbose regular expressions: https://docs.python.org/3/library/re.html#re.VERBOSE

I think it would be good to transform this into a verbose regex, since it's really heard to understand what's going on here.